.github/workflows/error-linter.yml

```diff
@@ -0,0 +1,55 @@
+name: code-error-linter
+on:
+  push:
+    branches:
+      - master
+      - main
+  pull_request:
+
+permissions:
+  contents: read
+  # Optional: allow read access to pull request. Use with `only-new-issues` option.
+  # pull-requests: read
+
+jobs:
+  golangci:
```

Comment:
Instead of separated workflow I would rather add this job to run-tests.yaml

Reply:
⚠️ currently we have step [`Run linter`](https://github.com/bitcoin-sv/spv-wallet/blob/0d59e794b5f7ce481ecb660640ad02bf9ba3e2d9/.github/workflows/run-tests.yml#L72) in run-tests.yaml job `test` and this step should be removed, because it would be better to have linters run as separated job.

Reply:
Added job to run-tests.yaml